driver-core/src/main/java/com/datastax/driver/core/DefaultResultSetFuture.java

```diff
@@ -82,14 +82,34 @@ public void onSet(Connection connection, Message.Response response, ExecutionInf
                                         // We'll add it back if CASSANDRA-5358 changes that behavior
                                         //if (scc.keyspace.equals(session.poolsState.keyspace))
                                         //    session.poolsState.setKeyspace(null);
-                                        session.cluster.manager.metadata.removeKeyspace(scc.keyspace);
+                                        final KeyspaceMetadata removed = session.cluster.manager.metadata.removeKeyspace(scc.keyspace);
+                                        if(removed != null) {
+                                            session.cluster.manager.executor.submit(new Runnable() {
```

Comment:
Same here: broadcasting of schema change events si done in the manager's executor so that it doesn't block the I/O thread that received the event notification.